src/index.ts

```diff
@@ -38,31 +45,51 @@ const rest = new REST({ version: '10' }).setToken(token!);
 
 async function main() {
   try {
-      console.log('Started refreshing application (/) commands.');
-
-      await rest.put(
-          Routes.applicationGuildCommands(clientID!, guildID!),
-          {
-              body: [
-                  Ping.info.toJSON(),
-                  Mock.info.toJSON()
-              ]
-          }
-      );
-
-      console.log('Successfully reloaded application (/) commands.');
+    console.log('Started refreshing application (/) commands.');
+
+    await rest.put(
+      Routes.applicationGuildCommands(clientID!, guildID!),
+      {
+        body: [
+          Ping.info.toJSON(),
+          Mock.info.toJSON(),
+          Cat.info.toJSON()
+        ]
+      }
+    );
+
+    console.log('Successfully reloaded application (/) commands.');
   } catch (error) {
-      console.error(error);
+    console.error(error);
   }
 }
 
-
-export const mockTargets = new Set();
-
 client.on('interactionCreate', async (interaction) => {
   if (!interaction.isCommand()) return;
 
   const { commandName } = interaction;
+  const userID = interaction.user.id;
+
+  if (!cooldown.has(commandName)) {
+    cooldown.set(commandName, new Map<string, number>());
+  }
+  const cooldownMap = cooldown.get(commandName)!;
+
+  if (cooldownMap.has(userID) && cooldownMap.get(userID)! > Date.now()) {
+    const cooldownRemaining = (cooldownMap.get(userID)! - Date.now()) / 1000;
+    await interaction.reply(`Please wait ${cooldownRemaining.toFixed(1)} seconds.`);
+    return;
+  }
+  cooldownMap.set(userID, Date.now() + cooldownTime);
+
+  for (const [commandName, cooldownMap] of cooldown) {
```

Comment:
Any reason why you need this? Seems like you don't really need this unless you intend on saving memory

Reply:
> Any reason why you need this? Seems like you don't really need this unless you intend on saving memory

I guess I interpreted one of the points of your issue as needing to do something like this. If you say its unnecessary then ill remove it.

" Automatically remove the user from the record when the rate limit expires for them. " is the reason for me adding this, although I can see how everything would work perfectly fine without it.